lib/SILGen/SILGenProlog.cpp

```diff
@@ -445,6 +451,8 @@ void SILGenFunction::emitProlog(const CaptureInfo &captureInfo,
 
   // Emit the capture argument variables. These are placed last because they
   // become the first curry level of the SIL function.
+  assert((captureInfo.hasBeenComputed() || cannotCapture(DC)) &&
```

Comment:
You can early return and skip the whole loop below if cannotCapture() is false.

Reply:
I don't think it'll make a real difference because the loop will exit very quickly anyway, and to be honest I don't trust `canCaptureFromParent()` enough to risk a subtle miscompile from it.